### ISSUE TITLE
[ENH] - consider consolidating conda-store and conda-server into one package

### ISSUE THREAD

**Issue by trallard** [conda-incubator/conda-store#911]
### Feature description

We currently have two packages in this repo, `conda-store` and `conda-store-server`, where the former is but a convenience package for the conda-store CLI.

This increases complexity, and users should ideally get the CLI and UI-based versions installed simultaneously (since there are no scenarios in which users would only install the CLI and use it without the server).

We should consolidate the two in a single package.

### Value and/or benefit

- Reduced complexity (what is conda-store? What is conda-store-server? which makes it hard for users to grasp conda-store as a whole).
- Reduced maintenance overhead: we would only release and maintain one package

### Anything else?

We would need to think of how to communicate this change to users.

**Comment by trallard** [comment]
@conda-incubator/conda-store team as we discussed this recently - it would be good to note any pros/cons/questions on your end.

**Comment by peytondmurray** [comment]
A note about timeline and how this change will be implemented: @smeragoel is currently enumerating user workflows that are important to capture for the forthcoming CLI. @soapy1 and I will take a few days to provide input about these workflows. When CLI work begins, that will be a good opportunity for consolidation.

**Comment by soapy1** [comment]
I've started looking into what needs to be done to make this happen. In particular, there are a few changes to be made:
* on the conda-store cli side
  * stop publishing the conda-store docker container
    *  there are a few conda-store docs/demos that depend on this container being available (and jupyter being set up in them)
  * remove all the `jupyter-*` dependencies from conda-store cli
* on the conda-store server side
  * it would be nice to be able to use the python api exposed in conda-store-server to de-duplicate code between the cli and public api
  * it would be nice to have a clear place for the cli module to live

A little bit of a rogue suggestion, but since we aren't really pushing any updates to the conda-store cli (until the new cli work begins) it might be nice to just drop/deprecate this package (stop building the conda-store packages and delete the `./conda-store` directory). 

Pros:
* we don't have to keep releasing this code that we are not making changes to
* users will still be able to use older version of the package 
* it will give use the opportunity to prepare the conda-store-server side of things to take a different approach at building the cli 

Cons:
* will not be able to provide bug fixes

**Comment by soapy1** [comment]
Few more tasks that need to be done before we can close this issue:

- [ ] (1) remove the conda-store module
- [ ] (2) rename conda-store-server to conda-store + move the new conda-store module one level up in the file structure
- [ ] (3) along with (2), we'll need to update the name of the conda-store-server package to conda-store + do all the comms that come with that
